src/main/java/dev/cdevents/events/ServiceDeployedCDEvent.java

```diff
@@ -0,0 +1,220 @@
+                "                  \"minLength\": 1\n" +
+                "                },\n" +
+                "                \"source\": {\n" +
+                "                  \"type\": \"string\"\n" +
+                "                }\n" +
+                "              },\n" +
+                "              \"additionalProperties\": false,\n" +
+                "              \"type\": \"object\",\n" +
+                "              \"required\": [\n" +
+                "                \"id\"\n" +
+                "              ]\n" +
+                "            },\n" +
+                "            \"artifactId\": {\n" +
+                "              \"type\": \"string\",\n" +
+                "              \"minLength\": 1\n" +
+                "            }\n" +
+                "          },\n" +
+                "          \"additionalProperties\": false,\n" +
+                "          \"type\": \"object\",\n" +
+                "          \"required\": [\n" +
+                "            \"environment\",\n" +
+                "            \"artifactId\"\n" +
+                "          ]\n" +
+                "        }\n" +
+                "      },\n" +
+                "      \"additionalProperties\": false,\n" +
+                "      \"type\": \"object\",\n" +
+                "      \"required\": [\n" +
+                "        \"id\",\n" +
+                "        \"type\",\n" +
+                "        \"content\"\n" +
+                "      ]\n" +
+                "    },\n" +
+                "    \"customData\": {\n" +
+                "      \"oneOf\": [\n" +
+                "        {\n" +
+                "          \"type\": \"object\"\n" +
+                "        },\n" +
+                "        {\n" +
+                "      \"type\": \"string\",\n" +
+                "      \"contentEncoding\": \"base64\"\n" +
+                "        }\n" +
+                "      ]\n" +
+                "    },\n" +
+                "    \"customDataContentType\": {\n" +
+                "      \"type\": \"string\"\n" +
+                "    }\n" +
+                "  },\n" +
+                "  \"additionalProperties\": false,\n" +
+                "  \"type\": \"object\",\n" +
+                "  \"required\": [\n" +
+                "    \"context\",\n" +
+                "    \"subject\"\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    /**
+     * @param subjectId
+     * sets the subject Id
+     */
+    public void setSubjectId(String subjectId) {
+        getSubject().setId(subjectId);
+    }
+
+    /**
+     * @param subjectSource
+     * sets the subject source
+     */
+    public void setSubjectSource(URI subjectSource) {
+        getSubject().setSource(subjectSource);
+    }
+
+    /**
+     * @param subjectType
+     * sets the subject type
+     */
+    public void setSubjectType(CDEventConstants.SubjectType subjectType) {
```

Comment:
Subject Type will be set to "service" for this event, method to set the type is not needed.